src/btree/bt_sync.c

```diff
@@ -325,6 +333,17 @@ __wt_sync_file(WT_SESSION_IMPL *session, WT_CACHE_OP syncop)
              */
             page = walk->page;
 
+            /*
+             * Skip clean pages, but need to make sure maximum transaction ID is always updated.
+             */
+            if (!__wt_page_is_modified(page)) {
+                if (((mod = page->modify) != NULL) && mod->rec_max_txn > btree->rec_max_txn)
+                    btree->rec_max_txn = mod->rec_max_txn;
+                if (mod != NULL && btree->rec_max_timestamp < mod->rec_max_timestamp)
```

Comment:
Minor nit: Is there a reason in the first `if` statement you did `mod->rec_max_txn > btree->rec_max_txn` and in the 2nd `if` statement you did `btree->rec_max_timestamp < mod->rec_max_timestamp`? It requires thought about why they're different directions? I suggest making them consistent with each other one way or the other.